tests/performance/test_bigquery_benchmarks.py

```diff
@@ -101,12 +121,29 @@ def test_taxi_trips_benchmark(
         ]
         assert len(expected_results) == len(actual_results)
         for expected_result, actual_result in zip(expected_results, actual_results):
-            # Assert individual keys so that test doesn't fail if new keys are added.
-            # Note: if this proves too fragile, consider enhancing logic to ignore extra nested keys and/or only check
-            # specific keys.
-            for expected_key in expected_result.keys():
-                assert expected_key in actual_result
-                assert expected_result[expected_key] == actual_result[expected_key]
+            description_for_error_reporting = (
+                f'{expected_result["expectation_config"]["expectation_type"]} result'
+            )
+            _recursively_assert_actual_result_matches_expected_result_keys(
+                expected_result, actual_result, description_for_error_reporting
+            )
+
+
+def _recursively_assert_actual_result_matches_expected_result_keys(
+    expected, actual, description_for_error_reporting
+):
+    """Only assert on keys that exist in the expected result, so that test is less fragile and doesn't incorrectly fail
```

Comment:
Maybe note explicitly that the "expected" input can be a subset of actual, and that this validates the contents of the expected subset, not the entire result set.

Reply:
Done.